File: Learning_Materials/Beginner_Level/Java/B6Scope.java
public class B5Scope {
  static void assignMethod() {
    int numInFunction = 5;
  }

  static int returnMethod() {
    int numInFunction = 5;
    return numInFunction;
  }

  public static void main(String[] args) {

    // in this lesson we will be talking about scope. Scope is the area in which
    // a variable is (visible) accessible. For example, if you declare (create) a
    // variable inside a if-else statement, it will only be accessible inside that
    // if-else statement.

    boolean bool = true;
    if (bool) {
      int numInIfStatement = 5; // creates the variable x and sets it to 5 in the If statement
      System.out.println(numInIfStatement); // this will work because x is accessible here
    } // the Curly brackects represent the end of the if statements scope

    System.pit.println(numInIfStatement); // this will not work because x is not accessible here

    // The same goes for Functions
    myMethod();
    System.out.println(numInFunction); // this will not work because numInFunction is not accessible here

    // However, there are two simple ways to get around this. The first is to
    // declare the variable outside of the function or if statement.
    int y = 5; // variable is outside of if statement
    if (z) {
      y -= 5; // modfies the variable y
    }
    System.out.println(y); // this will work because y is accessible here
    // The second way is to return the variable from the function
    System.out.print(returnMethod()); // this will work because the function returns the variable

    // This also goes for objects and classes, which we will cover next
  }
}

File: Learning_Materials/Beginner_Level/Python/B6Scope.py
# in this lesson we will be talking about scope. Scope is the area in which
# a variable is (visible) accessible. For example, if you declare (create) a
# variable inside a function, it will only be accessible inside that
# function.


def testfunction():
    x = 10


testfunction()
print(
    x
)  # this will cause an error, because x is not accessible outside of the function

# there are two ways around this
# 1. you can declare the variable outside of the function
# 2. you can return the variable from the function


def anotherFunction():
    x = 5
    return x


print(anotherFunction())  # this will print 5

# This also goes for objects and classes, which we will cover later

# FINAL NOTE! Python and java differ in how they handle scope.
# In java, variables create in if else statements and loops are subject to scope,
# and won't be visible (accessible) outside of there

bool = True
if bool:
    x = 10
print(x)  # this will cause an error in java, but not here, in python
